keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/VerificationRepositoryTests.kt

```diff
@@ -0,0 +1,151 @@
+package com.netflix.spinnaker.keel.persistence
+
+import com.netflix.spinnaker.keel.api.DeliveryConfig
+import com.netflix.spinnaker.keel.api.Environment
+import com.netflix.spinnaker.keel.api.Verification
+import com.netflix.spinnaker.keel.api.verification.VerificationContext
+import com.netflix.spinnaker.keel.api.verification.VerificationRepository
+import com.netflix.spinnaker.keel.api.verification.VerificationState
+import com.netflix.spinnaker.keel.api.verification.VerificationStatus
+import com.netflix.spinnaker.keel.api.verification.VerificationStatus.PASSED
+import com.netflix.spinnaker.keel.api.verification.VerificationStatus.RUNNING
+import com.netflix.spinnaker.keel.artifacts.DockerArtifact
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.EnumSource
+import strikt.api.expectCatching
+import strikt.assertions.isEqualTo
+import strikt.assertions.isNotNull
+import strikt.assertions.isNull
+import strikt.assertions.isSuccess
+
+abstract class VerificationRepositoryTests<IMPLEMENTATION : VerificationRepository> {
+
+  abstract fun createSubject(): IMPLEMENTATION
+
+  val subject: IMPLEMENTATION by lazy { createSubject() }
+
+  open fun VerificationContext.setup() {}
+
+  private data class DummyVerification(override val id: String) : Verification {
+    override val type = "dummy"
+  }
+
+  @Test
```

Comment:
Just curious, since you mentioned this in chat recently: should we be discussing switching to JUnit as the new standard, in your opinion? Prons/cons compared to minutests?

Reply:
Eh, I just find performance in IntelliJ bad when there are many/large nested lambdas (which is the entire basis of Minutest's API). I don't know that I feel strongly enough to say "this is what we should do" but I thought I'd experiment with just using JUnit here

Reply:
I wish we could positively blame that on the lambdas, but IntelliJ has been slow for me for all projects lately, so I don't know if that's the only reason. 🤷‍♂️  

In any case, I've no objections to using JUnit. I'm interested in learning more about how to model tests using this approach, since I understand there's no equivalent to the "build-up" approach we use with minutest nested contexts, which helps with reuse (but adds complexity). I'd appreciate if you can point me to examples as you experiment with it. 🙏

Reply:
Yeah, I'd definitely like to refactor some of the common data setup here. There are some subtle differences but I think a couple of factory methods could probably go a long way.